picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageDecoder
+import android.os.Build
+import android.os.Build.VERSION
+import android.util.TypedValue
+import androidx.annotation.DrawableRes
+import androidx.annotation.RequiresApi
+import okio.*
+import java.io.IOException
+import java.nio.ByteBuffer
+import kotlin.math.floor
+
+object BitmapUtils {
+
+    internal class ExceptionCatchingSource(delegate: Source?) : ForwardingSource(delegate!!) {
+        var thrownException: IOException? = null
```

Comment:
```suggestion
        var thrownException: IOException? = null

```

Reply:
Whats the suggestion here? Seems the same.